### PR TITLE
Added pre-launch BAM archive check for GPL Report trigger

### PR DESCRIPTION
**Pull request by victorskl**
Related #117

### CODE REVIEW

File: src/containers/Subject.js [hunk 1]
```diff
@@ -266,29 +266,92 @@ class Subject extends Component {
...
-        launchPadRowData = {
-          subject_id: subject.id,
-          ...data,
-        };
-      } catch (e) {
+      // NOTE: a bit of quick & dirty crude logic implementation here
+      // don't expect this to be stay longer after GPL ported into ICA
+      // as this should better be part of ICA Pipeline Workflow Automation like everyone else
+      // The need of on-demand "LaunchPad" feature is different thing -- i.e. to be considered
+      // better in Portal v2 revamp from ground up! Or, elsewhere dashboard. We shall see...
```
Comment (victorskl) [inline src/containers/Subject.js:279]: FYI @scwatts @reisingerf @williamputraintan -- I will push this through for now. Take it as just another prototype idea / MVP to discuss further later on. :)

File: src/containers/Subject.js [hunk 2]
```diff
@@ -266,29 +266,92 @@ class Subject extends Component {
...
+
+      if (launchGpl) {
+        try {
+          const init = {
+            headers: { 'Content-Type': 'application/json' },
+            body: {
+              subject_id: subject.id,
+            },
+          };
+          const data = await API.post('gpl', '', init);
```
Comment (williamputraintan) [inline src/containers/Subject.js:343]: Since this uses `lambda-url`, for now this would not be able to be triggered from React?

**Comment by williamputraintan** [comment]
Trying this locally and seems that it is `blocked by CORS policy`, I suppose this might also be the case when deployed.

**Review by williamputraintan — APPROVED** [review]

File: src/containers/Subject.js (hunk 2 above)
Comment (victorskl) [reply src/containers/Subject.js:343]: Umm, it will trigger as POST payload (i.e. data in `init`)

Comment (williamputraintan) [reply src/containers/Subject.js:343]: Ah, OK! Didn't realize that [AWS amplify](https://docs.amplify.aws/cli/auth/import/#import-an-existing-identity-pool) could invoke url with [AWS Signature Version 4 (SigV4)](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html)

Comment (victorskl) [reply src/containers/Subject.js:343]: Right! Yup, Amplify can do that.

**Comment by victorskl** [comment]
Merging...

**Pull request merged**